src/boot/mlang.ml

```diff
@@ -200,7 +200,8 @@ let resolve_id {normals; _} ident =
   | Some(ident') -> ident'
   | None -> empty_mangle ident
 
-(* TODO(vipa): this function is here since the current implementation has variables and constructors in the same namespace, it should be replaced by correct uses of resolve_id and resolve_con *)
+(* TODO(vipa): this function is here since the current implementation has variables and constructors in the same namespace, it should be replaced by correct uses of resolve_id and resolve_con
+   I am not sure what this is doing now, and if we can remove it. *)
```

Comment:
As mentioned during the meeting, this function does lookup with the assumption that variables and constructors live in the same namespace. `resolve_id` and `resolve_con` do lookup assuming they live in different namespaces. With the current approach we do nothing and keep `resolve_id_or_con`, but with what we talked about during the meeting each `resolve_id_or_con` and `delete_id_and_con` needs to be replaced with the corresponding specific `id` or `con` version. I can do this replacement as a PR to this PR after that change has been made.

Reply:
OK, good. I have now pushed an update that separates the name space between constructors and variables. Tests for MExpr passes, but not for MLang (most likely because of the above). Can you take a look and then create a patch to this PR? Thanks.